internal/xds/translator/listener.go

```diff
@@ -344,6 +344,25 @@ func findXdsHTTPRouteConfigName(xdsListener *listenerv3.Listener) string {
 	return ""
 }
 
+func buildTCPProxyHashPolicy(lb *ir.LoadBalancer) []*typev3.HashPolicy {
```

Comment:
can we reuse existing code from http ?

Reply:
There isn't a significant amount of code that can be shared between the two functions since they initialize distinct types of HashPolicy resources. We can consolidate them into a single buildHashPolicy function, however it will lead to varying return types and likely necessitate some type casting.

Reply:
cool, lets stick to this